src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPoolGroup.cs

```diff
@@ -156,8 +156,16 @@ internal DbConnectionPool GetConnectionPool(DbConnectionFactory connectionFactor
             if (null != _poolGroupOptions)
             {
                 DbConnectionPoolIdentity currentIdentity = DbConnectionPoolIdentity.NoIdentity;
+
                 if (_poolGroupOptions.PoolByIdentity)
                 {
+                    // There is no concept of Windows identity on non-Windows platform. Hence we 
+                    // cannot support Windows authentication.
+                    if (!ADP.IsWindows)
+                    {
+                        throw new PlatformNotSupportedException(SR.GetString(SR.ADP_FeatureNotSupportedOnNonWindowsPlatform, DbConnectionStringKeywords.IntegratedSecurity));
```

Comment:
SR.GetString has been added to SqlClient and is used everywhere in the library. So for consistency I went ahead with it.